### DISCUSSION TITLE
How to indicate a QInput is mandatory / required?

### DISCUSSION THREAD

**Discussion by eugenevk** [quasarframework/quasar#11951] [General - Components / Directives / etc] [answered]
I want to indicate to the user that an input is mandatory. I can do so by adding e.g. an asterisk (*) at the end of the input label, but I want to make it more clear, e.g. by displaying the asterisk in red. How can I accomplish that?
I also tried setting the input class to a yellow background, but then I get this:

![image](https://user-images.githubusercontent.com/19785727/148636356-9b255987-e3ed-4d16-8c5c-14ea9f900855.png)
As you can see the yellow color is not nicely applied to the field: on the left is a white space and the yellow color covers the borders. How can I solve that?

My code:
```
<q-input
  input-class="bg-yellow-1"
  label="Name *"
  lazy-rules
  outlined
  :rules="[ val => val && val.length > 0 || 'Name is mandatory' ]"
  v-model=...
/>
```

**Comment by MilosPaunovic** [comment]
By Material design [specs](https://material.io/components/text-fields), this is the way to mark input field as required:

> To indicate that a field is required, display an asterisk (*) next to the label text and mention near the form that asterisks indicate required fields.

And if you are set on changing the colors of your input, take a look at the `bg-color` and `label-color` props.
[https://codepen.io/milospaunovic/pen/ExwpqmK?editors=1010](https://codepen.io/milospaunovic/pen/ExwpqmK?editors=1010
)

**Comment by pdanpdan** [comment]
Use the label slot for the red *
Apply the bg color on qinput if you want it all over
Make codepens with what you have tried if you want more help for that kind of problems

**Comment by metalsadman** [comment]
you can override the `q-field__label` class. ie 
```css
.q-field__label:after
    content: ' *'
    color: red
```

**Comment by smolinari** [comment, marked as answer]
Because it's so easy.

https://codepen.io/smolinari/pen/qBPMWwg?editors=1010

Scott

**Reply by eugenevk** [reply]
Beautiful! I'm pretty new to Quasar, after having used VueJS for quite a while, and love Quasar's capabilities and what comes out of the box. Thanks @smolinari @metalsadman @MilosPaunovic @pdanpdan

**Comment by janssen-dev** [comment]
If you plan to reuse rules a lot, I recommend defining them in a separate file like this:

_ValidationRules.js_
```
export default {
	required(val) {
		return !!val || "Field is required"
	}
}
```

_template_
```
<q-input
  input-class="bg-yellow-1"
  label="Name *"
  lazy-rules
  outlined
  :rules="[required]"
  v-model=...
/>
```

_script_
```
import ValidationRules from "../util/ValidationRules"

export default {
	name: "...",
	setup() {
		return {
			...ValidationRules,
			//...
		}
	}
}
```

So you only have a nice looking and clean `:rules="[required]"` :)
